Store version1

yermolenko-d opened the pull request.

Commit:
PageSwitch class created. Added references to Web.config.

Commit:
Finished with Pages. Finally.

Commit:
Created nice URL for List of all cakes.
RouteConfig.cs

Commit:
Updated Bootstrap to version 4.1(?).
Created Nice View for List action via Bootstrap.
Created partial view.

Commit:
Created routing by categories.
RouteConfig.cs
CakeController.cs
CakeListView.cs

Commit:
Done with Navigation System and finished (I hope so) with bootstrap

Commit:
Fixed bug with amount of pages

Commit:
Created Class Basket

Commit:
Basket logic added. Created Basket controller. Created button for adding in a basket.

Commit:
Changed Basket Controller, created CakeListView Model for View.
Created View Index for Basket Controller (Not finished).

Commit:
Finished with Basket. Finally.

Commit:
Cleared Basket Controller from method GetBasket by adding a Model Binder. Controller logic divided from Creating a basket logic.
Now any method that works with creating a basket can just add a basket in its signature.

Commit:
Minor fixes. Added a tab of basket on a navbar.

YaroslavKuchuk commented on Store.MVC/App_Start/RouteConfig.cs:
```diff
@@ -13,6 +13,32 @@ public static void RegisterRoutes(RouteCollection routes)
         {
             routes.IgnoreRoute("{resource}.axd/{*pathInfo}");
 
+            routes.MapRoute(null, "",
```
Why  it is so hard?

YaroslavKuchuk commented on Store.MVC/Content/bootstrap-reboot.css:
```diff
@@ -0,0 +1,331 @@
+}
+
+[tabindex="-1"]:focus {
+  outline: 0 !important;
```
just advice - !important is a bad style of css-ing. But it is not in scope of project.

yermolenko-d replied:
> just advice - !important is a bad style of css-ing. But it is not in scope of project.

I thought about to change all styles to make it looks like my homepage. But I'll do it at the last point i think. Now back-end functional is more important to make.

YaroslavKuchuk commented on Store.MVC/Controllers/BasketController.cs:
```diff
@@ -0,0 +1,66 @@
+
+        public RedirectToRouteResult AddToBasket(Basket basket, int id, string urlReturn)
+        {
+            Cake cake = repository.Cakes
```
Do we have a chance to create a service tier in app?

YaroslavKuchuk commented on Store.MVC/Controllers/BasketController.cs:
```diff
@@ -0,0 +1,66 @@
+        {
+            return PartialView(basket);
+        }
+        // GET: Basket
```
no comments in code.

If you are using GitHub or something else - what is the purpose to use comments?

yermolenko-d replied:
Done. And by the way thanks for explanation.

YaroslavKuchuk commented on Store.MVC/Controllers/CakeController.cs:
```diff
@@ -18,18 +20,32 @@ public CakeController(ICakeRepository repository)
+            CakeListView model = new CakeListView
+            {
+                Cakes = cakeRepository.Cakes
+                   .Where(c => category == null || c.Category == category)
```
As I mentioned above - create a service including repository as dependency and then move all LINQ methods to it

yermolenko-d replied:
> As I mentioned above - create a service including repository as dependency and then move all LINQ methods to it

Well, I've been trying to do this. But it takes so much elements in its signature if I will make it as a independed method. So thats a problem...

YaroslavKuchuk reviewed (commented):
try to fix my comments please

Commit:
Created order sendings via Ninject.
ShippingDetails.cs
EmailOrderProcessor.cs
IOrderProcessor.cs

added new stylesheet for errors: ErorStyleSheet.css
added link in _Layout.cshtml

Commit:
Minor fixes in Checkout view.

Commit:
-Created _AdminLayout.cshtml for admin view.
-Added Admin Cotroller, that allows us to manage products.
-Created Partial View for index Method in AdminController.
-Minor Fixes with shipping.

Commit:
Deadline is still burning but I'm never give up!
So, I've finished with changing products on admin panel. Finally.
Also created validation via models metadata.
Funny comments included.(Cake.cs)

So much time lost in case of inattentive naming. But it works. And it's good i think.

By the way, NavigationController have been cleared from that trash comments.

Commit:
Created Action Create. It was easily that i imagined.

In Edit.cshtml I'm using an overload of method Html.BeginForm.

Commit:
Finished with delete action.
AdminController.cs
EFCakeReository.cs
ICakeRepositry.cs

Commit:
Authentitication added.

Commit:
Changed style of all project

Commit:
Unnecessary Bootstrap files deleted

Commit:
Added navigation links

Commit:
-Final minor fixes with design

-Some Unit-Tests added
